src/wallet/home_cards/deferred_deep_link/parser.rs

```diff
@@ -0,0 +1,286 @@
+use crate::prelude::*;
+use base64::engine::general_purpose::URL_SAFE_NO_PAD;
+use base64::Engine;
+
+#[async_trait::async_trait]
+pub trait DeferredDeepLinkParser: Send + Sync {
+    async fn parse(&self, encoded_value: String) -> Result<HomeCards>;
+}
+
+pub struct Parser {
+    gateway_client: GatewayClient,
+}
+
+impl Parser {
+    pub fn new(gateway_client: GatewayClient) -> Self {
+        Self { gateway_client }
+    }
+}
+
+#[async_trait::async_trait]
+impl DeferredDeepLinkParser for Parser {
+    async fn parse(&self, encoded_value: String) -> Result<HomeCards> {
+        let decoded = self.decode(encoded_value)?;
+        self.transform_onboarding_deep_link_value(decoded).await
+    }
+}
+
+impl Parser {
+    fn decode(
+        &self,
+        encoded_value: impl AsRef<str>,
+    ) -> Result<OnboardingDeepLinkValue> {
+        let decoded_value_json_bytes = URL_SAFE_NO_PAD
+            .decode(encoded_value.as_ref())
+            .map_err(|e| {
+                println!("{}", e);
+                CommonError::DeferredDeepLinkInvalidValueFormat
+            })?;
+
+        serde_json::from_slice::<OnboardingDeepLinkValue>(
+            decoded_value_json_bytes.as_ref(),
+        )
+        .map_err(|_| CommonError::DeferredDeepLinkDecodingFailed)
+    }
+}
+
+#[cfg(test)]
+mod tests_decode {
+    use super::*;
+
+    #[allow(clippy::upper_case_acronyms)]
+    type SUT = Parser;
+
+    fn make_sut() -> SUT {
+        SUT::new(GatewayClient::new(
+            Arc::new(MockAntenna::new_always_failing()),
+            NetworkID::Stokenet,
+        ))
+    }
+
+    #[test]
+    fn decode_correct() {
+        let sut = make_sut();
+        let encoded_value = "eyJtZXRob2QiOiJtb2JpbGUiLCJkYXBwX3JlZmVycmVyIjoiYWNjb3VudF9yZHgxMjh5Nmo3OG10MGFxdjYzNzJldnoyOGhyeHA4bW4wNmNjZGRrcjd4cHBjODhoeXZ5bnZqZHdyIiwic3BlY2lhbF9kYXBwIjoicmFkcXVlc3QifQ";
+        let result = sut.decode(encoded_value).unwrap();
+        assert_eq!(result, OnboardingDeepLinkValue::sample());
+    }
+
+    #[test]
+    fn decode_invalid_value_format() {
+        let sut = make_sut();
+        let encoded_value = "invalid format";
+        let result = sut.decode(encoded_value).unwrap_err();
+        assert_eq!(result, CommonError::DeferredDeepLinkInvalidValueFormat);
+    }
+
+    #[test]
+    fn decode_decoding_failed() {
+        let sut = make_sut();
+        let encoded_value = "e30";
+        let result = sut.decode(encoded_value).unwrap_err();
+        assert_eq!(result, CommonError::DeferredDeepLinkDecodingFailed);
+    }
+}
+
+impl Parser {
+    async fn transform_onboarding_deep_link_value(
+        &self,
+        value: OnboardingDeepLinkValue,
+    ) -> Result<HomeCards> {
```

Comment:
After going through this function, I don't see any error that is being thrown, so this does not need to be a Result.